src/context/FormEventActions/FormEventActionsFactory.ts

```diff
@@ -0,0 +1,16 @@
+import { FormEventActions } from "./FormEventActions";
+import { IFormEventActions } from "./IFormEventActions";
+
+export class FormEventActionsFactory {
+    public static buildFormEventActions(formContext: Xrm.FormContext, xrm: Xrm.XrmStatic): IFormEventActions {
+        const areV9MethodsAvailable = !!formContext
+                                    && !!formContext.data
+                                    && !!formContext.data.entity
+                                    && !!formContext.data.entity.save;
+        return areV9MethodsAvailable

```

Comment:
Can we split that event more? This looks to me that it will grow in a weird way.

It currently has multiple purposes:
1. The handling of UI stuff, like notifications.
2. The handling of the `save`
3. The handling of attributes.

MS's form context V9 has this architecture:
https://docs.microsoft.com/en-us/dynamics365/customer-engagement/developer/clientapi/clientapi-form-context#formcontext-object-model

I think we should stick to this, or to something similar.